### PR TITLE
Document user personas

### LINKED ISSUE #757: Create key personas to tackle

### PR DESCRIPTION
**Pull request by pavithraes**
Fixes #757 

## Description

This pull request:

- Documents the final user personas created for conda-store

## Pull request checklist

- [x] Did you test this change locally?
- [ ] Did you update the documentation (if required)?
- [ ] Did you add/update relevant tests for this change (if required)?

## Additional information

🤝 Add `@smeragoel`, `@jaimergp`, and `@trallard` as co-authors while merging

### CODE REVIEW

File: docusaurus-docs/community/design/user-personas.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+- Troubleshooting:
+    - Understand errors through the error message displayed
+    - Look at conda-store’s documentation & GH issues/discussions, search for the error
+    - Contact team-members or internal support
+    - Open an issue/discussion on GH
+
+#### Pain points or biggest challenges
+
+- Envs need to be compliant with company standards
+- Build envs with libraries from internal mirrors
```
Comment (kcpevey) [inline docusaurus-docs/community/design/user-personas.md:64]: This is in contrast with "ability to use any package/version out there" https://github.com/conda-incubator/conda-store/pull/773/files#diff-99d5523c0e67c61630b4cfb694883c9282df0074dcd29f29078be74bdb1efc2fR48

File: docusaurus-docs/community/design/user-personas.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+### User: Dani
+
+| Role | Reports to | User group|
+|-------|-------|-------|
+Research Scientist and Coordinator | PI | Admin |
+
+#### User story
+
+As a research coordinator, I manage the logistics and admin-tasks of multiple research projects, ensuring that the tools/dependencies used are accessible, reproducible and standardised across projects.
```
Comment (kcpevey) [inline docusaurus-docs/community/design/user-personas.md:80]:
```suggestion
As a research coordinator, I manage the logistics and admin-tasks of multiple research projects, ensuring that the tools/dependencies used are accessible, reproducible and standardized across projects.
```

File: docusaurus-docs/community/design/user-personas.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+- Discovery:
+    - OSS Community, conferences, peers
+    - Tries out conda-store locally and finds it useful
+- Onboarding:
+    - Documentation, self-exploration
+- General Use:
+    - Setup, deployment using the conda-store docs, within current Lab infra
+    - Add team members, create default namespaces and environments, referring to the docs
+    - Onboard team members to the tool with demonstrations
+    - Track and limit resource utilisation
```
Comment (kcpevey) [inline docusaurus-docs/community/design/user-personas.md:97]:
```suggestion
    - Track and limit resource utilization
```

File: docusaurus-docs/community/design/user-personas.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+- Collaboration
+    - Share relevant artifacts (lockfiles?) with clients
+- Troubleshooting
+    - Docs, issue tracker
+
+#### Pain points or biggest challenges
+
+- Sometimes conda environments break while working, adding overhead
+- Environments are not always reproducible when I share them
+- Using shared artifacts w/ the application should be as easy as possible (think installers/executables)
```
Comment (kcpevey) [inline docusaurus-docs/community/design/user-personas.md:150]:
```suggestion
- Using shared artifacts with the application should be as easy as possible (think installers/executables)
```

**Review by kcpevey — COMMENTED** [review]
This is looking great! 

I think we are missing a junior dev persona. This person is fresh out of bootcamp/newly hired/intern/ and their supervisor doesn't want to burden them with creating envs. They don't need to create envs, but are given one via conda-store by their supervisor.

File: docusaurus-docs/community/design/user-personas.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    - A team member gives them a walkthrough
+    - Documented Tutorials
+    - Intuitiveness of the UI
+- General Use:
+    - Env creation from GUI, YAML, Lockfile
+    - Version control
+    - Using the env in Jupyter Notebook (local / cloud)
+    - Ability to use any package/version out there
+    - Access to special conda channels
+    - Envs for GPU-powered data analysis and modelling
```
[the remark this answers is not in the document]
Comment (pavithraes) [reply docusaurus-docs/community/design/user-personas.md:50]: US English :)

File: docusaurus-docs/community/design/user-personas.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+#### User story
+
+I want to quickly create a stable and reproducible environment for analyzing data and creating dashboards that I can share with my team to verify and build on my work.
+
+#### Tools they need to do their job
+
+- Libraries - data reader/writer (arrow), data processing (pandas), numerical computing (numpy), ML (xgboost / pytorch), dataviz and dashboarding (mpl / streamlit), job schedulers (Airflow)
+- IDE - JupyterLab/Notebook (hence, extensions)
+- Platform - Local computer, org JupyterHub on Cloud (Nebari?)
```
Comment (pavithraes) [inline docusaurus-docs/community/design/user-personas.md:32]:
```suggestion
- Platform - Local computer, org JupyterHub on Cloud (potentially Nebari)
```

File: docusaurus-docs/community/design/user-personas.md [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+    - Download artifacts like docker images for use in other (prod?) machines
+- Troubleshooting:
+    - Understand errors through the error message displayed
+    - Look at conda-store’s documentation & GH issues/discussions, search for the error
+    - Contact team-members or internal support
+    - Open an issue/discussion on GH
+
+#### Pain points or biggest challenges
+
+- Envs need to be compliant with company standards
```
Comment (pavithraes) [inline docusaurus-docs/community/design/user-personas.md:63]:
```suggestion
- Environments need to be compliant with company standards, for example: approved conda channels and approved package versions
```

File: docusaurus-docs/community/design/user-personas.md [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+|-------|-------|-------|
+Research Scientist and Coordinator | PI | Admin |
+
+#### User story
+
+As a research coordinator, I manage the logistics and admin-tasks of multiple research projects, ensuring that the tools/dependencies used are accessible, reproducible and standardised across projects.
+
+#### Tools they need to do their job
+
+In addition to Alia’s tools, cloud/hpc tooling
```
[the remark this answers is not in the document]
Comment (pavithraes) [reply docusaurus-docs/community/design/user-personas.md:84]: This user would interface with GCP for example, and use the gcloud CLI. They may also dig into the k8s logs and such. I'll add some clarification. :)

File: docusaurus-docs/community/design/user-personas.md [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+# User groups and personas
+
+The personas used for informing user research studies and future improvements to conda-store.
+
+## User groups 🏘️
+
+At a high-level, there are two key user groups:
+
+- **General user:** Create, edit, use environments for their day-to-day work
+- **Admin-type user:** In addition to the general use-case mentioned above, admin-type users also **setup** conda-store for their team, create various namespaces, add/update users and their permissions, set default packages/versions for their team, etc.
```
[the remark this answers is not in the document]
Comment (pavithraes) [reply docusaurus-docs/community/design/user-personas.md:14]: Good catch, I think that was left behind from discussions/notes. I'll update it.

File: docusaurus-docs/community/design/user-personas.md (hunk 1 above)
Comment (pavithraes) [reply docusaurus-docs/community/design/user-personas.md]: I don't think so? We meant this as: internal mirrors are one of the various places we'd want to install packages from, right?

File: docusaurus-docs/community/design/user-personas.md [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+    - Download artifacts like docker images for use in other (prod?) machines
+- Troubleshooting:
+    - Understand errors through the error message displayed
+    - Look at conda-store’s documentation & GH issues/discussions, search for the error
+    - Contact team-members or internal support
+    - Open an issue/discussion on GH
+
+#### Pain points or biggest challenges
+
+- Envs need to be compliant with company standards
```
[the remark this answers is not in the document]
Comment (pavithraes) [reply docusaurus-docs/community/design/user-personas.md]: added more info :)

**Comment by pavithraes** [comment]
Preview link to personas, for quick reference: https://deploy-preview-773--conda-store.netlify.app/community/design/user-personas

**Review by smeragoel — APPROVED** [review]

**Pull request merged**